addons/website_slides/controllers/main.py

```diff
@@ -643,18 +684,107 @@ def _slide_channel_prepare_review_values(self, channel):
 
         return values
 
-    @http.route('/slides/channel/enroll', type='http', auth='public', website=True)
-    def slide_channel_join_http(self, channel_id):
-        # TDE FIXME: why 2 routes ?
+    @http.route('/slides/<int:channel_id>/invite', type='http', auth='public', website=True, sitemap=False)
+    def slide_channel_invite(self, channel_id, partner_id=False, invite_hash=False):
+        """ This route is included in the invitation link in email to join the course. It is the main entry point
+        for sharing course to attendees or to invite them directly.
+
+        It acts as a dipatcher, here are the supported use cases:
+            - If the current user (public or logged) can access the course, redirects to the course page
+            - If no user is logged and the public user has no access to the course, redirects to login page,
+            then redirects in turn to this route to be dispatched again, once the user is logged, according
+            to its access rights.
+            - If a logged user has not access to the course, redirects to the main page with all courses, with an
+            invite_state in the url that will trigger an error message to explain the user what is the problem.
+
+        :param channel_id: The id of the course the user is invited to. Do not use <model> in the route
+                           instead, otherwise an error 403 could be returned if the (public) user has no
+                           access to the record.
+
+        :return: redirects to the relevant route, see above.
+        """
+
+        def default_routing(channel=False, has_rights=False, invite_state=''):
```

Comment:
I first thought of a more explicit name,
```suggestion
        def slides_redirect(channel=False, has_rights=False, invite_state=''):
```

But now I think we'd be better with a `def redirect_channel(channel)` and a `def redirect_slides_main(invite_state)` because here we are going from a specific failure case to pass arguments to be parsed to redirect to the right page :recycle: , it seems unnecessary as we don't use the specific channel in cases of failure.